slsim/lens.py

```diff
@@ -502,7 +524,50 @@ def point_source_magnitude_molet(self, band, time, **kwargs_molet):
 
         # TODO: in what format should be the 2d source profile be stored (as it is time- and wavelength dependent)
         # TODO: do we create full light curves (and save it in cache) or call it each time
-        return 0
+
+        # we can use this to get intrinsic brightness of a source at image observation
+        # time. Here, we are storing source profile as dictionary of observation time
+        # and magnitude but need to decide what format molet needs. I checked with molet
+        #  and it says format should be in the form given below.
+        source_profile = [
+            {
+                "time": list(self.source.lightcurve_time),
+                "signal": list(
+                    self._point_source_magnitude(
+                        band=band, lensed=False, time=self.source.lightcurve_time
+                    )
+                ),
+            }
+        ]
+        # using source profile, kappa, gamma, kappa_star, image_observed_time molet
+        # should return lightcurve of each images. The lightcurve can be a interpolated
+        # function or a dictionary of observation time and magnitudes in specified band
+        # as given below.
+        # This molet_output is temporary. Once we call molet, this will be actual molet
+        # output.
+        self.molet_output = [
```

Comment:
@nkhadka21 I would code it to first check whether this molet output exists, and only if not,, then perform all the parts, and else simply call the interpolation function.
also, please make this variable a private variable